lightning/src/ln/peers/transport.rs

```diff
@@ -0,0 +1,404 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+/// Abstraction for the transport layer as described by Bolt #8 [https://github.com/lightningnetwork/lightning-rfc/blob/master/08-transport.md]
+
+use bitcoin::secp256k1::{SecretKey, PublicKey};
+
+use ln::peers::conduit::Conduit;
+use ln::peers::handler::{IOutboundQueue, ITransport, PeerHandleError};
+use ln::peers::handshake::PeerHandshake;
+use ln::{wire, msgs};
+use ln::wire::{Encode, Message};
+
+use util::ser::{Writeable, VecWriter};
+use util::logger::Logger;
+use std::ops::Deref;
+
+/// Interface used by Transport to interact with a handshake object
+pub trait IPeerHandshake {
```

Comment:
This probably needs a rename ("I"-prefix seems to me to imply internal?) if its going to be public, as well as likely a much more thorough set of docs to explain what it does.

Reply:
Sure. I didn't find an existing pattern for interface traits in the codebase. In previous jobs, interfaces had "I" prefixes/suffixes so the types were identifiable. Let me know what you would prefer and I can do the rename.

Reply:
Hmm, right, I think there's less chance of confusion in Rust since things either need to be explicitly generic across a trait or (at least once we upgrade to newer rust) carry the `dyn` identifier. I'm not sure a prefix is required, I obviously misread the intent. In any case, does this need to be pub, or should it be pub only in fuzztarget mode/for testing? I'm not sure what the specific use-case is for the API, maybe @arik-so had one?